### DISCUSSION TITLE
Saving CMS pages from admin back-end and VCS

### DISCUSSION THREAD

**Discussion by arvislacis** [wintercms/winter#297] [Ideas]
Maybe there are some historical or technical reasons but I was thinking about some interesting behavior currently happening when saving CMS pages from admin back-end UI:

1. If page has PHP code section then `<?php` and `?>` tags are added to the PHP code section (if they original doesn't exist).
2. Also when page is saved then any trailing new-lines are removed from file.

Both of these behaviors conflicts with VCS like git and/or different coding standarts, for example, when source code files always should have trailing new-line. Also, for example, Winter CMS doc examples mostly everywhere shows CMS page structure with PHP code section without `<?php` / `?>` tags...

So I was thinking - are there some particular reason for this? I am mostly referring to the 1st point... Maybe there should be additional configuration options for this, for example, which would not add `<?php` / `?>` and another one for trailing new-lines?

Or I better should try to achieve this with my own custom plugin functionality?

**Comment by bennothommo** [comment]
I'm not quite sure on the reasoning on the first point - I can only assume it was to make it clear that the content block was a PHP script, but there could very well be other reasons why it is done.

On the second point, I believe that's because the content blocks are run through `trim()` on save to remove any extra spacing.

**Reply by LukeTowers** [reply]
With the first point I personally add them myself to all the files I work on, I think it makes it easier for syntax highlighting to work correctly.

**Reply by LukeTowers** [reply]
The docs examples could always be updated

**Reply by arvislacis** [reply]
@bennothommo @LukeTowers Thanks for your answers.

In my opinion the problem is consistency and/or configuration options. Probably both of these things could be made configurable in `../config/cms.php` file, like, for example, `convertLineEndings` option is.

Anyway, I will try to look at the option how to solve this with custom plugin functionality or something...

**Reply by LukeTowers** [reply]
I'd be fine with just making it so files always end with a newline by default personally
